src/pages/CardPage.tsx

```diff
@@ -1,6 +1,102 @@
+import { useEffect, useState } from "react";
 import { useParams } from "react-router-dom";
+//import Footer from "@/components/Footer"; // Already renders, figuring out if I need to add a declarions file later and adjust main rendering. not prio, just annoying to see
+import { cn } from "@/lib/utils";
+import "./CardPage.css";
+//import React from "react";
+
+interface Card {
+  id: number;
+  name: string;
+  descriptionRaw: string;
+  attack: number;
+  cost: number;
+  health: number;
+  flavorText: string;
+  artistName: string;
+  cardCode: string;
+  collectible: boolean;
+  set: string;
+  assets: {
+    gameAbsolutePath: string;
+    fullAbsolutePath: string;
+  }[];
+  formats: string[];
+}
 
 export default function CardPage() {
   const { code } = useParams();
-  return <div>This is the card details page for card number {code}</div>;
+  const [card, setCard] = useState<Card | null>(null);
+
+  useEffect(() => {
+    async function fetchCard() {
+      try {
+        const response = await fetch(`http://127.0.0.1:3000/api/v1/cards/${code}`);
```

Comment:
This should be set to the heroku server because it's being deployed to production.